src/helpers/game_utils.ts

```diff
@@ -22,12 +23,13 @@ interface GroupMatchResults {
  * @param ignoredVideoIds - List of Youtube video IDs of songs to ignore
  * @returns a list of songs, as well as the number of songs before the filter option was applied
  */
-async function getFilteredSongList(guildPreference: GuildPreference, ignoredVideoIds?: Array<string>): Promise<{ songs: QueriedSong[], countBeforeLimit: number }> {
+async function getFilteredSongList(guildPreference: GuildPreference, ignoredVideoIds?: Array<string>, alternatingGender?: GENDER): Promise<{ songs: QueriedSong[], countBeforeLimit: number }> {
     let result: Array<QueriedSong>;
     if (!guildPreference.isGroupsMode()) {
+        const gender = guildPreference.getSQLGender() === GENDER.ALTERNATING ? [GENDER.MALE, GENDER.FEMALE] : guildPreference.getSQLGender().split(",");
```

Comment:
Remove getSQLGender and replace it with getGender that returns an array of genders